### ISSUE TITLE
Variable is set but never used for pushes

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#1730]
Today I had a very basic problem, but it was difficult to see.

```php
function test($userId)
{
    $userData = [];
    $response = [ 'user' => $userData ];

    $userData['id'] = $userId;

    return $response;
}

var_dump(test(123));
// Expected: [ 'user' => [ 'id' => 123 ] ]
// Received: [ 'user' => [] ] (but it is normal)
```

The code above, _in theory_, should return an array containing the user's data and the id passed by parameter. But, PHP does not treat arrays as Object and, if the reference is not explicit (like `$response = [ 'user' => &$userData ]`), the push/set would have no effect.

My suggestion is to try to present this as a possible flaw: _the array has been updated, but its new value has never been used_. And to correct it, the ideal would be: either use reference or redefine the variable after push/set, as below.

```php
function test($userId)
{
    $userData = [];
    $userData['id'] = $userId;

    $response = [ 'user' => $userData ];

    return $response;
}
```

**Comment by ktomk** [comment]
You might be interested in doing static code analysis, e.g. this should be in the set of Phpstorm (and reported against it as this is what they do for variables) and perhaps run psalm or phpstan who also do similar things.

This does not take away that while _writing_ code, these "potential" flaws are most often a hindrance, so if you could elaborate how this would be specifically helpful within EA inspections would be great.

**Comment by ea-inspections-team** [comment]
:grin: that looks familiar, thanks for reporting @rentalhost. We'll look into it.

@ktomk : we have an inspection for write-only variables, where this patter wold we be a great addition.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @rentalhost. Looks like PhpStorm now reports this case as well.

**Comment by ktomk** [comment]
muchas gracias. :+1:
